airbyte-cdk/python/airbyte_cdk/sources/streams/concurrent/concurrent_stream.py

```diff
@@ -2,46 +2,157 @@
+        :param max_workers:
+        :return:
+        """
+        return ConcurrentStream(
+            partition_generator=LegacyPartitionGenerator(stream),
+            max_workers=max_workers,
+            name=stream.name,
+            json_schema=stream.get_json_schema(),
+            availability_strategy=AvailabilityStrategyLegacyAdapter(stream, stream.availability_strategy)
+            if stream.availability_strategy
+            else None,
+            primary_key=stream.primary_key,
+            cursor_field=stream.cursor_field,
+        )
+
+    def __init__(
+        self,
+        partition_generator: PartitionGenerator,
+        max_workers: int,
+        name: str,
+        json_schema: Mapping[str, Any],
+        availability_strategy: Optional[AvailabilityStrategy],
+        primary_key: Optional[Union[str, List[str], List[List[str]]]],
+        cursor_field: Union[str, List[str]],
+    ):
+        self._stream_partition_generator = partition_generator
+        self._max_workers = max_workers
+        self._threadpool = concurrent.futures.ThreadPoolExecutor(max_workers=self._max_workers, thread_name_prefix="workerpool")
+        self._name = name
+        self._json_schema = json_schema
+        self._availability_strategy = availability_strategy
+        self._primary_key = primary_key
+        self._cursor_field = cursor_field
+
     def read(
         self,
         cursor_field: Optional[List[str]],
         logger: logging.Logger,
         slice_logger: SliceLogger,
         internal_config: InternalConfig = InternalConfig(),
     ) -> Iterable[StreamData]:
-        raise NotImplementedError
+        logger.debug(f"Processing stream slices for {self.name} (sync_mode: full_refresh)")
+        total_records_counter = 0
+        futures = []
+        queue = Queue()
+        record_sentinel = object()
+        partition_sentinel = object()
+        partition_generator = ConcurrentPartitionGenerator(queue, partition_sentinel)
+        partition_reader = PartitionReader(queue, record_sentinel)
+
+        # Submit partition generation tasks
+        futures.append(
+            self._threadpool.submit(partition_generator.generate_partitions, self._stream_partition_generator, SyncMode.full_refresh)
+        )
+
+        TIMEOUT_SECONDS = 300
+
+        finished_partitions = False
+        while record_or_partition := queue.get(block=True, timeout=TIMEOUT_SECONDS):
+            if record_or_partition == partition_sentinel:
+                finished_partitions = True
+            elif finished_partitions and record_or_partition == record_sentinel:
+                break
+            elif is_record(record_or_partition):
+                yield record_or_partition.stream_data
+                if AbstractStream.is_record(record_or_partition.stream_data):
+                    total_records_counter += 1
+                    if internal_config and internal_config.is_limit_reached(total_records_counter):
+                        return
```

Comment:
@girarda this is where we use `_limit` parameters from internal config.
The idea behind it was to limit number of reads, not records returned, but since we have all partitions in the pool it is kinda useless, so I propose to ignore this functionality for now and completely remove it from AbstractStream and here.